### DISCUSSION TITLE
Reset all other fields when a field is updated

### DISCUSSION THREAD

**Discussion by marcebdev** [react-hook-form/react-hook-form#10087] [Q&A]
I have a top-level field `val1` as part of a form, when it is changed I want to reset the values for all other fields, how can I achieve this?

If I do 
```
const watchVal1 = useWatch({ name: 'val1', control })

useEffect(() => {
  reset({ val1: watchVal1 }, { keepDefaultValues: true })
}, [watchVal1])
```
the value doesn't change and it breaks all my other `useWatch` hooks even if I were to use a `setValue` after the reset

using `reset` inside the `register('val1')`'s onChange causes an infinite call stack which makes sense. Is there really no way to better way to batch reset fields?

is `resetField` or `setValue` for each individual field in the `onChange` really the best or only way?

**Comment by leapful** [comment]
@marcebdev 

Please provide a codesandbox url to re-produce the issue. Usually, calling `reset()` in `onChange` callback will not cause infinite re-rendering.
